libhirte/src/node.c

```diff
@@ -3,20 +3,68 @@
 #include "../include/bus/peer-bus.h"
 #include "../include/bus/systemd-bus.h"
 #include "../include/bus/user-bus.h"
+#include "../include/common/common.h"
 #include "../include/node.h"
 #include "./common/dbus.h"
 
+static int node_signal_handler(
+                sd_event_source *event_source, UNUSED const struct signalfd_siginfo *si, UNUSED void *userdata) {
+        // Do whatever cleanup is needed here.
+
+        sd_event *event = sd_event_source_get_event(event_source);
+        sd_event_exit(event, 0);
+        return 0;
+}
+
+static int node_setup_signal_handler(sd_event *event) {
```

Comment:
The node and orchestrator handler are mostly the same. I know its not ideal due to the dependency on another PR (#69), but I think it would be great to move it to the `service/shutdown.c` and generalize it. Then the method could be sth like
```C
bool service_register_shutdown_signal(sd_event *event_loop, int sig);
```
and pass the `sd_event* event_loop` as userdata here:
```C
// signal_shutdown would be a kind of wrapper function
sd_event_add_signal(event, NULL, sig, signal_shutdown, event_loop);
```

The caller would be able to define for which signals he wants to setup the shutdown handling, e.g. in the binaries.

Reply:
@engelmi So let's merge your PR and we will rebase ours and change according to this

Reply:
Great idea!  I'll wait for your PR and then rebase.

Reply:
#69 is merged :heavy_check_mark:

Reply:
Your suggestion is great.  In the interest of time, I suggest I add a "TO DO" comment for this and come back to it later, so I can help with other more important tasks.

Reply:
I consolidated the event loop signal setup into the shutdown service code.  I simplified it a bit so that this is specifically for handling shutdown type signals (SIGTERM and SIGINT), with ```sd_event_add_signal()``` doing the event loop shutdown.